website/upgrade-guides/0-14.html.markdown

```diff
@@ -0,0 +1,303 @@
+---
+layout: "guides"
+page_title: "Upgrading to Terraform v0.14"
+sidebar_current: "upgrade-guides-0-14"
+description: |-
+  Upgrading to Terraform v0.14
+---
+
+# Upgrading to Terraform v0.14
+
+Terraform v0.14 is a major release and so it includes some small changes in
+behavior that you may need to consider when upgrading. This guide is intended
+to help with that process.
+
+The goal of this guide is to cover the most common upgrade concerns and
+issues that would benefit from more explanation and background. The exhaustive
+list of changes will always be
+[the Terraform Changelog](https://github.com/hashicorp/terraform/blob/v0.14/CHANGELOG.md).
+After reviewing this guide, we recommend reviewing the Changelog to check for
+specific notes about less-commonly-used features. This guide is also not
+intended as an overview of the new features in Terraform v0.14. This release
+includes other enhancements that don't need any special attention during
+upgrade, but those are described in the changelog and elsewhere in the
+Terraform documentation.
+
+This guide focuses on changes from v0.13 to v0.14. Terraform supports upgrade
+tools and features only for one major release upgrade at a time, so if you are
+currently using a version of Terraform prior to v0.13 please upgrade through
+the latest minor releases of all of the intermediate versions first, reviewing
+the previous upgrade guides for any considerations that may be relevant to you.
+
+In particular, Terraform v0.14 no longer includes the `terraform 0.13upgrade`
+command for automatically inserting
+[provider requirements](/docs/configuration/provider-requirements.html)
+into existing modules, and the automatic mechanisms to upgrade legacy provider
+references in the Terraform state. You will need to successfully complete a
+`terraform apply` at least once under Terraform v0.13 before upgrading an
+existing configuration to Terraform v0.14.
+
+-> If you run into any problems during upgrading that are not addressed by the
+information in this guide, please feel free to start a topic in
+[The Terraform community forum](https://discuss.hashicorp.com/c/terraform-core),
+describing the problem you've encountered in enough detail that other readers
+may be able to reproduce it and offer advice.
+
+Upgrade guide sections:
+
+* [Before You Upgrade](#before-you-upgrade)
+* [Provider Dependency Lock File](#provider-dependency-lock-file)
+* [Concise Terraform Plan Output](#concise-terraform-plan-output)
+* [Sensitive Values in Plan Output](#sensitive-values-in-plan-output)
+
+## Before You Upgrade
+
+When upgrading between major releases, we always recommend ensuring that you
```

Comment:
I think we should be more prescriptive on this recommendation (in all upgrade guides): the most common upgrade issues I've run across in several major upgrades is largely due to folks not having a no-op plan before upgrading.

Reply:
I agree with being prescriptive, but what I already wrote here feels prescriptive to me... :thinking:

Can you say a little more about what might be missing in this section to make it more prescriptive?

Reply:
Ahh I found your other comment below that already says more.

The following paragraph here already states that you _must_ have run `terraform apply` with 0.13 before upgrading, and that's not only guidance but a hard requirement: if someone doesn't do that then `terraform init` will return an error about legacy providers. So my instinct is to leave this unchanged and let the second paragraph be the _must_, because saying that you _must_ run `terraform plan` and then that you also _must_ run `terraform apply` seems redundant, given that `terraform apply` incorporates `terraform plan`.

Reply:
Indeed, I think you figured out my meaning much better than I'd stated it initially: My thought is that the recommendation to have a clear plan is what comes _first_, and it's not uncommon for folks (ok, me) to only remember the first thing they read (I'm a terrible skimmer). I like your idea.